aodntools/timeseries_products/gridded_timeseries.py

```diff
@@ -0,0 +1,262 @@
+        VoI_ndepths = xr.DataArray(np.full(1, 0), coords=[[time_0]], dims=['TIME'])
+
+        ## group nc by individual timestamps
+        VoI_grouped = nc.groupby('TIME')
+
+
+        for timestamp, (name, group) in enumerate(VoI_grouped):
+            time = [name]
+            n_depths = int(len(group[VoI]))
+
+            if n_depths >= 2:
+                VoI_values = list(group[VoI].values)
+                depth = list(group.DEPTH.values)
+                ## sort depths
+                depth, VoI_values = sort_depths(depth, VoI_values)
+
+                ## check for max separation
+                depth_mask = get_depth_mask(depth_bins=depth_bins, depths=depth, max_separation=max_separation)
+                ## do the interpolation
+                VoI_gridded = np.interp(depth_bins, depth, VoI_values, left=np.nan, right=np.nan)
+                ## set masked depth bins to zero
+                VoI_gridded = VoI_gridded * depth_mask
+                VoI_gridded[VoI_gridded == 0] = np.nan
+
+            else:
+                VoI_gridded = np.full((depth_bin_len, 1), np.nan)
+
+            VoI_temp = xr.DataArray(VoI_gridded.reshape(depth_bin_len, 1), coords=[time, depth_bins],
+                                    dims=['TIME', 'DEPTH'])
+            VoI_ndepths_temp = xr.DataArray([n_depths], coords=[time], dims=['TIME'])
+
+            VoI_binned = xr.concat([VoI_binned, VoI_temp], dim='TIME')
+            VoI_ndepths = xr.concat([VoI_ndepths, VoI_ndepths_temp], dim='TIME')
+
+    VoI_interpolated = xr.Dataset({VoI: VoI_binned.astype(float),
+                                   VoI + '_count': VoI_ndepths.astype(int)})
+
+    ## drop the very first record as it is dummy
+    VoI_interpolated = VoI_interpolated.where(VoI_interpolated.TIME >= time_min, drop=True)
+
+    ## Add lat/lon as scalar variables
+    VoI_interpolated = VoI_interpolated.assign(LONGITUDE = longitude_mean,
+                                               LATITUDE = latitude_mean)
+
+    ## transpose dimensions to make CF compliant
+    VoI_interpolated = VoI_interpolated.transpose('TIME', 'DEPTH')
+
+    ## get the variables attribute dictionary
+    with open(TEMPLATE_JSON) as json_file:
+        attr_dictionary = json.load(json_file)
+
+    variable_attribute_dictionary = attr_dictionary['_variables']
+    global_attribute_dictionary = attr_dictionary['_global']
+
+    ## set variable attributes
+    varlist = list(VoI_interpolated.variables)
+    add_variable_attribute = {}
+    variable_attributes = TStools.set_variableattr(varlist, variable_attribute_dictionary, add_variable_attribute)
+    time_units = variable_attributes['TIME'].pop('units')
+    time_calendar = variable_attributes['TIME'].pop('calendar')
+    for variable in varlist:
+        VoI_interpolated[variable].attrs = variable_attributes[variable]
+
+    ## set global attributes
+    timeformat = '%Y-%m-%dT%H:%M:%SZ'
+    VoI_interpolated.attrs.update(global_attributes)
```

Comment:
I'm a purist - if you are overwriting, just overwrite - the update method is just noise here. 

Also, it's very hard to know what is being updated  because the argument is enormous and the variable names you created are big (e.g. global_attribute_dictionary).
Since their scope is short - you only use them here - a short name is handy.

finally, it's much easier to update a key at a time, and/or to create a dictionary before hand:

```
dt_start_str = pd.to_datetime(VoI_interpolated.TIME.values.min()).strftime(timeformat);
...
updated_gattrs = {'file_version': gatt_dict['file_version'],
                  'source_file': file_name,
                  'time_coverage_start': dt_start_str,
...
```

Reply:
:+1: 
`.update()` is fine with me, but I think some simple re-formatting (shortening lines and removing extra spaces) would help make this more readable... e.g.
```python
    VoI_interpolated.attrs.update({
        'file_version': global_attribute_dictionary['file_version'],
        'source_file': file_name,
        'time_coverage_start': pd.to_datetime(VoI_interpolated.TIME.values.min()).strftime(timeformat),
￼        # etc...
    })
```